pkg/controller/oam/traitdefinition/traitdefinition_controller.go

```diff
@@ -0,0 +1,89 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package traitdefinition
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1alpha2 "github.com/crossplane/crossplane/apis/oam/v1alpha2"
+)
+
+// TraitDefinitionReconciler reconciles a TraitDefinition object
+type TraitDefinitionReconciler struct {
+	client.Client
```

Comment:
Better add a `Client` field. Embed is only good for _inheritance_ scenario.

Reply:
This is auto-generated by kubebuilder.... What tools do u use to scaffolding?